src/main/java/ttt/Board.java

```diff
@@ -2,149 +2,108 @@
 
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 
 public class Board {
 
-    private String[] grid = new String[9];
-    private String status;
-    private String emptyCell;
-    private String winningMark;
+    private Marks[] board = new Marks[9];
+    private final Marks emptyCell;
```

Comment:
I wouldn't worry about using this `emptyCell` instance variable.
I'd just use Marks.EMPTY wherever I need it. It's just as descriptive and abstract "enough".